### ISSUE TITLE
support client specifying acceptable media type as resource path extension

### ISSUE THREAD

**Issue by nisavid** [nisavid/bedframe#17]
the goal here is to handle requests for URIs of the form `resource.ext`, where
- _resource_ is an existing resource URI,
- _ext_ is a recognized file extension with a corresponding _mediatype_, and
- the requested web method supports the inferred _mediatype_ for its response content.

in such a scenario, the requested web method should be treated as only supporting the inferred _mediatype_ for its response content.

in other words, if the client's acceptable media ranges (according to the `Accept` header) include the inferred _mediatype_, then the response content type should be that _mediatype_.  if not, then the response should be a `NoAcceptableMediaType` exception (in HTTP, `406 Not Acceptable`).

Bedframe should provide a default mapping of file extensions to media types, as well as some means to customize this mapping.
